packages/flutter/lib/src/material/calendar_date_picker.dart

```diff
@@ -27,11 +27,20 @@ import 'theme.dart';
 
 const Duration _monthScrollDuration = Duration(milliseconds: 200);
 
-const double _dayPickerRowHeight = 42.0;
+// Current M2 implementation is not compliant with the M2 specification.
+// Instead of a 42 pixels row height it should be 40 with a 2 pixels inner padding.
+// See: https://m2.material.io/components/date-pickers#specs.
+const double _dayPickerRowHeightM2 = 42.0;
+// For M3, row height is 48 pxiels with 4 pixels inner padding.
+// See: https://m3.material.io/components/date-pickers/specs#2d53890e-a08f-4c63-a0d9-abd9e95b4245.
+const double _dayPickerRowHeightM3 = 48.0;
+
 const int _maxDayPickerRowCount = 6; // A 31 day month that starts on Saturday.
 // One extra row for the day-of-week header.
-const double _maxDayPickerHeight = _dayPickerRowHeight * (_maxDayPickerRowCount + 1);
-const double _monthPickerHorizontalPadding = 8.0;
+const double _maxDayPickerHeightM2 = _dayPickerRowHeightM2 * (_maxDayPickerRowCount + 1);
+const double _maxDayPickerHeightM3 = _dayPickerRowHeightM3 * (_maxDayPickerRowCount + 1);
+
+const double _monthPickerHorizontalPadding = 12.0;
```

Comment:
It seems to me that this line caused the google failures, as reverting this line alone makes the tests pass. I guess that this line affects both the portrait mode and the landscape mode. What surprises me is that the public golden tests don't contain any failures for landscape mode. Do you have an idea?

Reply:
Thanks for isolating this line :pray: 
I pushed a commit to apply the new value only when M3 and portrait (initially this PR contained both portrait and landscape fixes, I missed this line when I split this PR). 

> What surprises me is that the public golden tests don't contain any failures for landscape mode. Do you have an idea?

Oh, I checked and there are no golden test for landscape mode.
I will add one in my next PR which will target landscape only.